cmd/cdi-importer/importer.go

```diff
@@ -121,51 +106,90 @@ func main() {
 		os.Exit(1)
 	}
 	if source == controller.SourceNone {
-		if contentType == string(cdiv1.DataVolumeKubeVirt) {
-			createBlankImage(imageSize, availableDestSpace, preallocation, volumeMode, filesystemOverhead)
-			preallocationApplied = preallocation
-		} else {
-			errorEmptyDiskWithContentTypeArchive()
+		err := handleEmptyImage(contentType, imageSize, availableDestSpace, preallocation, volumeMode, filesystemOverhead)
+		if err != nil {
+			klog.Errorf("%+v", err)
+			os.Exit(1)
+		}
+
+	} else {
+		exitCode := handleImport(source, contentType, volumeMode, imageSize, filesystemOverhead, preallocation)
+		if exitCode != 0 {
+			os.Exit(exitCode)
 		}
+	}
+}
+
+func handleEmptyImage(contentType string, imageSize string, availableDestSpace int64, preallocation bool, volumeMode v1.PersistentVolumeMode, filesystemOverhead float64) error {
+	var preallocationApplied bool
+
+	if contentType == string(cdiv1.DataVolumeKubeVirt) {
+		createBlankImage(imageSize, availableDestSpace, preallocation, volumeMode, filesystemOverhead)
+		preallocationApplied = preallocation
 	} else {
-		waitForReadyFile()
-		klog.V(1).Infoln("begin import process")
+		errorEmptyDiskWithContentTypeArchive()
+	}
+
+	err := importCompleteTerminationMessage(preallocationApplied)
+	return err
+}
+
+func handleImport(
+	source string,
+	contentType string,
+	volumeMode v1.PersistentVolumeMode,
+	imageSize string,
+	filesystemOverhead float64,
+	preallocation bool) int {
+	klog.V(1).Infoln("begin import process")
 
-		ds = newDataSource(source, contentType, volumeMode)
-		defer ds.Close()
+	ds := newDataSource(source, contentType, volumeMode)
+	defer func() {
+		ds.Close()
+	}()
 
-		processor := newDataProcessor(contentType, volumeMode, ds, imageSize, filesystemOverhead, preallocation)
-		err = processor.ProcessData()
+	processor := newDataProcessor(contentType, volumeMode, ds, imageSize, filesystemOverhead, preallocation)
+	waitForReadyFile()
+	err := processor.ProcessData()
 
+	if err != nil {
+		klog.Errorf("%+v", err)
+		if err == importer.ErrRequiresScratchSpace {
+			return common.ScratchSpaceNeededExitCode
+		}
+		err = util.WriteTerminationMessage(fmt.Sprintf("Unable to process data: %+v", err))
```

Comment:
please don't remove the err.Error(), I added it it fixes an issue: https://github.com/kubevirt/containerized-data-importer/pull/2111

Reply:
Sorry, I did play with the format, temporary added some more verbose logging. so this was removed by mistake, reverting that part

Reply:
done